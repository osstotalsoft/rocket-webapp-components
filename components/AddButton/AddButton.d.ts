import { ButtonProps } from '@material-ui/core/Button';

export interface AddButtonProps extends Omit<ButtonProps, 'color'> {
  /**
  * The title of button.
  */
  title?: string
  /**
  * Callback fired when a "click" event is detected.
  */
  onClick?: () => void
  /**
  * @default "false"
  * If true, button will be disabled.
  */
  disabled?: boolean
  /**
  * @default "theme"
  * The color of button.
  */
  color?: "primary" |
          "info" |
          "theme" |
          "themeNoBackground" |
          "themeWithBackground" |
          "success" |
          "warning" |
          "danger" |
          "rose" |
          "white" |
          "simple" |
          "defaultNoBackground" |
          "primaryNoBackground" |
          "infoNoBackground" |
          "successNoBackground" |
          "warningNoBackground" |
          "dangerNoBackground" |
          "roseNoBackground"
  /**
  *  @default "medium"
  * The size of button. 
  */
  size?: "medium" | "small"
  /**
  *  @default "small"
  * The fontSize of button's content.
  */
  fontSize?: "inherit" | "small" | "medium" | "large"
  /**
  * Override or extend the styles applied to the component
  */
  customClass?: string
  /**
  * @default "false"
  * If true, the keyboard focus ripple will be disabled. 
  */
  disableFocusRipple?: boolean
  /**
  * If true, the ripple effect will be disabled.
  */
  disableRipple?: boolean
  /**
  * @default "false"
  * If given, uses a negative margin to counteract the padding on one side 
  * (this is often helpful for aligning the left or right side of the icon 
  * with content above or below, without ruining the border size and shape).
  */
  edge?: "start" | "end" | "false"
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/add-button
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/AddButton/AddButton.js
 * 
 * Bit component used:
 * 
 * - IconButton: https://bit.dev/totalsoft_oss/react-mui/icon-button
 * 
 */
export default function AddButton(props: AddButtonProps): JSX.Element;