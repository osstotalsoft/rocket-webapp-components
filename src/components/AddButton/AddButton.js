import React from "react";
import PropTypes from "prop-types";
import AddIcon from '@material-ui/icons/Add';
import CustomIconButton from '../IconButton';


const AddButton = ({ title, onClick, disabled, fontSize, ...rest }) => (
    <CustomIconButton aria-label="Add" onClick={onClick} disabled={disabled} tooltip={title} {...rest}>
        <AddIcon fontSize={fontSize} />
    </CustomIconButton>
)

AddButton.defaultProps = {
    color: "theme",
    fontSize: "small",
    size: "medium"
};

AddButton.propTypes = {
    /**
   * The title of button.
   */
    title: PropTypes.string,
    /**
    * Callback fired when a "click" event is detected.
    */
    onClick: PropTypes.func,
    /**
   * If true, the button will be disabled.
   */
    disabled: PropTypes.bool,
    /**
   * The color of button.
   */
    color: PropTypes.oneOf([
        "primary",
        "info",
        "theme",
        "themeNoBackground",
        "themeWithBackground",
        "success",
        "warning",
        "danger",
        "rose",
        "white",
        "simple",
        "defaultNoBackground",
        "primaryNoBackground",
        "infoNoBackground",
        "successNoBackground",
        "warningNoBackground",
        "dangerNoBackground",
        "roseNoBackground",
        "themeWithGradient"
    ]),
    /**
      * The size of button.
      */
    size: PropTypes.oneOf([
        "small",
        "medium"
    ]),
    /**
    * The fontSize of button's content.
    */
    fontSize: PropTypes.oneOf([
        "inherit",
        "small",
        "medium",
        "large"
    ])
};

export default AddButton;