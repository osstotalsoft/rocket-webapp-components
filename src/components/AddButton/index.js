import AddButton from "./AddButton";
export default AddButton;