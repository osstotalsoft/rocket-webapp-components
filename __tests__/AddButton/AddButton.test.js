import React from 'react';
import { mount } from "enzyme";
import AddButton from '../../components/AddButton/AddButton';

describe('AddButton', () => {
    it('the button is disabled', () => {
        const wrapper = mount(<AddButton
            title="Title"
            onClick={() => { }}
            disabled={true}
            fontSize="small"
            size="small"
        />);

        expect(wrapper.prop('disabled')).toBe(true)
    });

    it('the button is enabled', () => {
        const wrapper = mount(<AddButton
            title="Title"
            onClick={() => { }}
            disabled={false}
            fontSize="small"
            size="small"
        />);

        expect(wrapper.prop('disabled')).toBe(false)
    });

    it('the sended props are the same as the expected ones', () => {
        const wrapper = mount(<AddButton
            title="Title"
            onClick={() => { }}
            disabled={false}
            fontSize="small"
            size="small"
        />);

        expect(wrapper.prop('disabled')).toBe(false)
        expect(wrapper.prop('fontSize')).toBe('small')
        expect(wrapper.prop('size')).toBe('small')
    });

    it('the button is expected to be called', () => {
        const onChangeMock = jest.fn()
        const wrapper = mount(<AddButton
            title="Title"
            onClick={onChangeMock}
            disabled={false}
            fontSize="small"
            size="small"
        />);

        wrapper.find('Button').simulate('click');
        expect(onChangeMock).toHaveBeenCalledTimes(1);
    });
});