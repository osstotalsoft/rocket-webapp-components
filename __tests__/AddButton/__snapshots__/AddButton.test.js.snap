// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton snapshot 1`] = `
<span
  aria-describedby={null}
  className=""
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Title"
>
  <button
    aria-label="Add"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-button-1 makeStyles-theme-17 Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
    disabled={true}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
  </button>
</span>
`;
